src/main/java/com/epam/idea/core/repository/IdeaRepository.java

```diff
@@ -16,4 +16,14 @@
 	 */
 	@Query("select i from Idea i where i.author.id = ?1")
 	List<Idea> findByUserId(Long userId);
+
+	/**
+	 * Return a list of ideas which belongs to the user with given id,
+	 * or an empty list if the user has no ideas.
+	 *
+	 * @param tagId The id of the tag.
+	 * @return All the ideas of the tag.
+	 */
+	@Query("select i from Idea i inner join i.relatedTags t WHERE t.id IN (?1)")
```

Comment:
1. U can simplify this query:

```sql
select t.relatedIdeas from Tag t where t.id = ?1
```
2.Be consistent: if u want to use upper case for SQL keywords so every word have to be in upper case like this

```sql
SELECT i FROM Idea i INNER JOIN i.relatedTags t WHERE t.id IN (?1)
```
and vice versa.
3. Add integration test for this custom method.